### ISSUE TITLE
Question about CPL

### ISSUE THREAD

**Issue by Creator-wei** [TorchSSL/TorchSSL#66]
I have a question about CPL, in the code:models/flexmatch/flexmatch_utils.py line 44~46.

`44 mask = max_probs.ge(p_cutoff * (class_acc[max_idx] / (2. - class_acc[max_idx]))).float()  # convex
45 # mask = max_probs.ge(p_cutoff * (torch.log(class_acc[max_idx] + 1.) + 0.5)/(math.log(2) + 0.5)).float()  # concave
46 select = max_probs.ge(p_cutoff).long()`

I understand in the article is that in different iterations, the multiplied coefficients of thresholds are different, so as to achieve the purpose of filtering better samples. But in actual code

`46 select = max_probs.ge(p_cutoff).long()`

But I see that select is used for the calculation of class_acc[] in the next iteration for the data selection of unlabed, that is, it plays the role of filtering data in the next iteration.
In other words, the thresholds for screening for each class are not multiplied by coefficients. Here, the class_acc[] I understand is the coefficient that thresholds needs to be multiplied by.

More specifically, I don't quite understand whether the adjustment of the learning parameters in different iterations is done by adjusting the threshold coefficient in flexmatch? Or is the value optimized by adjusting the loss to optimize the model parameters?

**Comment by crush-kunkun** [comment]
i have the same question have the same question with you.uestion with you.the flex threshold  is only used for loss ,not for  Pseudo Labeling.
